### PR TITLE
{Containerapp} Set test region default to northcentralusstage

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Containerapp

**Review by yanzhudd — APPROVED** [review]

File: src/containerapp/azext_containerapp/tests/latest/custom_preparers.py [hunk 1]
```diff
@@ -25,9 +25,16 @@ def __init__(self, name_prefix='aks', location='eastus2euap', aks_name='my-aks-c
     def create_resource(self, name, **kwargs):
         group = self._get_resource_group(**kwargs)
         try:
-            self.live_only_execute(self.cli_ctx, f'az aks create --resource-group {group} --name {self.infra_cluster} --enable-aad --generate-ssh-keys --enable-cluster-autoscaler --min-count 4 --max-count 10 --node-count 4')
+            aks_location = self.location
+            arc_location = self.location
+
+            if self.location.__contains__('stage'):
+                aks_location = "southcentralus"
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/custom_preparers.py:32]: could we use centralus

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py [hunk 2]
```diff
@@ -15,22 +15,28 @@
...
-from .common import TEST_LOCATION
+from .common import TEST_LOCATION, STAGE_LOCATION
 from .utils import create_containerapp_env, prepare_containerapp_env_for_app_e2e_tests
 
 TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
 
 
 class ContainerappIdentityTests(ScenarioTest):
+    def __init__(self, *arg, **kwargs):
+        super().__init__(*arg, random_config_dir=True, **kwargs)
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:26]: blank line?

File: src/containerapp/azext_containerapp/tests/latest/utils.py [hunk 3]
```diff
@@ -34,7 +37,7 @@ def prepare_containerapp_env_for_app_e2e_tests(test_cls):
 
 def create_containerapp_env(test_cls, env_name, resource_group, location=None):
     logs_workspace_name = test_cls.create_random_name(prefix='containerapp-env', length=24)
-    logs_workspace_id = test_cls.cmd('monitor log-analytics workspace create -g {} -n {} -l {}'.format(resource_group, logs_workspace_name, TEST_LOCATION)).get_output_in_json()["customerId"]
+    logs_workspace_id = test_cls.cmd('monitor log-analytics workspace create -g {} -n {} -l eastus'.format(resource_group, logs_workspace_name)).get_output_in_json()["customerId"]
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/utils.py:40]: why do you hard code the location here?

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py [hunk 4]
```diff
@@ -77,44 +78,17 @@ def test_containerapp_e2e(self, resource_group):
         ])
 
 
-    # TODO rename
-    @AllowLargeResponse(8192)
-    @ResourceGroupPreparer(location="eastus2")
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py:82]: why remove the test

Comment (Greedygre) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_scenario.py:82]: There is another test with same name test_container_acr, the test case is duplicate.

File: src/containerapp/azext_containerapp/tests/latest/utils.py (hunk 3 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/tests/latest/utils.py:40]: Will revert it.

File: src/containerapp/azext_containerapp/tests/latest/custom_preparers.py (hunk 1 above)
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/custom_preparers.py:32]: should we just check the location is northcentralusstage

File: src/containerapp/azext_containerapp/tests/latest/test_containerappjob_crud.py [hunk 5]
```diff
@@ -122,8 +122,10 @@ def test_containerapp_manualjob_private_registry_port(self, resource_group):
         image_source = "mcr.microsoft.com/k8se/quickstart:latest"
 
         env_id = prepare_containerapp_env_for_app_e2e_tests(self)
-
-        self.cmd(f'acr create --sku basic -n {acr} -g {resource_group} --admin-enabled')
+        acr_location = TEST_LOCATION
+        if acr_location == STAGE_LOCATION:
+            acr_location = "eastus"
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerappjob_crud.py:127]: normalized first and compare

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_env_commands.py [hunk 6]
```diff
@@ -550,7 +553,11 @@ def test_containerapp_env_update_custom_domains(self, resource_group):
     @ResourceGroupPreparer(location="northeurope")
     @live_only()  # passes live but hits CannotOverwriteExistingCassetteException when run from recording
     def test_containerapp_env_internal_only_e2e(self, resource_group):
-        self.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+        # network is not available in North Central US (Stage), if the TEST_LOCATION is "northcentralusstage", use eastus as location
+        location = TEST_LOCATION
+        if location == "northcentralusstage":
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_env_commands.py:558]: use SATGE_LOCATION?

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_up_on_connected_env.py [hunk 7]
```diff
@@ -132,7 +135,10 @@ def test_containerapp_up_on_arc_auto_install_extension_e2e(self, resource_group,
         self.assertIsNotNone(custom_location_id)
         env_list = self.cmd(f'containerapp connected-env list -g {resource_group}').get_output_in_json()
         self.assertEqual(env_list[0]["extendedLocation"]["name"].lower(), custom_location_id.lower())
-        self.assertEqual(env_list[0]["location"], connected_cluster["location"])
+        if connected_cluster["location"] == 'eastus2euap':
+            self.assertEqual(env_list[0]["location"], STAGE_LOCATION)
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_up_on_connected_env.py:139]: >STAGE_LOCATION [](http://example.com/codeflow?start=54&length=14)

TEST_LOCATION

**Review by Juliehzl — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]
